### ISSUE TITLE
Compilation doesn't fail when referenced assemblies target mismatched .net versions

### ISSUE THREAD

**Issue by scionwest** [dotnet/roslyn#5001]
**TL;DR**
If a project references a  different version of the .net framework that the referenced assembly targets, it won't actually build (as expected). Roslyn does not give any indication that the compilation actually failed, rather it tells you everything is good to go. If you run the app, Visual Studio fails to start it because of errors while none exist in the error list. 

**Reproducing it**
I have a [repository with a solution](https://github.com/scionwest/RoslynCompilationIssue) that reproduces this. You can reproduce it with the following steps as well. 

1. Create a new Class Library, using .Net 4.5.2. 
2. Create a new Application (I've experienced it with both WinForms and WPF), using .Net 4.5 and add a reference to the class library. 
3. Create a new instance of the Class1 Type that came with the class library inside of the Application's Program.cs class.
3. Create a Unit Test project, using .Net 4.5 and add a reference to the WPF app. 

Compile the app. You will see that it builds without any issues. Now try and run the app. It'll fail to start. 

**Expected**
A compiler error in the error list when the target framework used by a referenced assembly is not usable by the project using the mismatched referenced assembly

**Actual**
No indication is given that the referenced assembly is preventing the project from properly building.

**Notes**

I noticed when I remove the Unit Test project, the compiler will sometimes generate a compiler error telling me it can't find the class library namespace, and then the error disappears. When I have the unit test project included in the solution, i do not ever see the error message.

There are also times I am simply told "The .exe reference can not be found" in the error list. This is because the unit test project references it, and the .exe project can not build due to the reference differentiation.

**Edit**
It does mention it in the output, however I believe this should be in the error list.

**Comment by jaredpar** [comment]
@scionwest thanks for reporting this issue.

Does a build error show up in the output window when this behavior occurs?  If so then it's likely a bug in how the error gets translated to the error list.

**Comment by scionwest** [comment]
The output only provides me with a warning and not an error as I would expect.

> C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3274: The primary reference "C:\Users\Sully\Source\Repos\RoslynCompilationIssue\DotNet452Library\bin\Debug\DotNet452Library.dll" could not be resolved because it was built against the ".NETFramework,Version=v4.5.2" framework. This is a higher version than the currently targeted framework ".NETFramework,Version=v4.5".

**Comment by scionwest** [comment]
@jaredpar The output only provides me with a warning and not an error as I would expect.

C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3274: The primary reference "C:\Users\Sully\Source\Repos\RoslynCompilationIssue\DotNet452Library\bin\Debug\DotNet452Library.dll" could not be resolved because it was built against the ".NETFramework,Version=v4.5.2" framework. This is a higher version than the currently targeted framework ".NETFramework,Version=v4.5".

Looking at the warning though, I'm wondering if this is actually an MSBuild issue, and not so much a Roslyn issue.

**Comment by lgolding** [comment]
@heejaechang: The fact that the warning doesn't show up in the Error List is a bug.

We will open a separate issue on the question of whether this should be a warning or an error.

**Comment by lgolding** [comment]
I reported the request for MSB3274 to the MSBuild team. I will let you know their response.

**Comment by jaredpar** [comment]
@lgolding is this related to the following? 

> https://github.com/Microsoft/msbuild/issues/173

**Comment by heejaechang** [comment]
due of this bug #4179

Workspace seems not setting compilation version number correctly
